<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ImageSpecialEffectsQtClass</class>
 <widget class="QMainWindow" name="ImageSpecialEffectsQtClass">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>821</width>
    <height>578</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ImageSpecialEffectsQt</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>817</width>
      <height>521</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>缩放</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSlider" name="horizontalSlider">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="minimum">
            <number>3</number>
           </property>
           <property name="maximum">
            <number>200</number>
           </property>
           <property name="value">
            <number>10</number>
           </property>
           <property name="tracking">
            <bool>false</bool>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_4">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>恢复原图大小</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLabel" name="label">
         <property name="minimumSize">
          <size>
           <width>640</width>
           <height>480</height>
          </size>
         </property>
         <property name="text">
          <string>No Image</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="text">
          <string>打开图像...</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_3">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>另存为图像...</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_13">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>undo</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_10">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>黑白化</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_2">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>灰度化</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_5">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>色彩平衡...</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnTextImage">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>转字符画...</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_7">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>去除亮度</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_6">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>边缘检测</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_8">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>模糊</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_9">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>黑白边缘平滑</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_11">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Salience</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_12">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>lab</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>821</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="imagespecialeffectsqt.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>ImageSpecialEffectsQtClass</receiver>
   <slot>openFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>814</x>
     <y>106</y>
    </hint>
    <hint type="destinationlabel">
     <x>765</x>
     <y>143</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>ImageSpecialEffectsQtClass</receiver>
   <slot>saveFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>814</x>
     <y>135</y>
    </hint>
    <hint type="destinationlabel">
     <x>790</x>
     <y>248</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>ImageSpecialEffectsQtClass</receiver>
   <slot>doGray()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>814</x>
     <y>261</y>
    </hint>
    <hint type="destinationlabel">
     <x>776</x>
     <y>290</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>ImageSpecialEffectsQtClass</receiver>
   <slot>updateSize(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>209</x>
     <y>49</y>
    </hint>
    <hint type="destinationlabel">
     <x>215</x>
     <y>70</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>ImageSpecialEffectsQtClass</receiver>
   <slot>undoZoom()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>686</x>
     <y>52</y>
    </hint>
    <hint type="destinationlabel">
     <x>738</x>
     <y>48</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_5</sender>
   <signal>clicked()</signal>
   <receiver>ImageSpecialEffectsQtClass</receiver>
   <slot>openBalanceDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>814</x>
     <y>290</y>
    </hint>
    <hint type="destinationlabel">
     <x>775</x>
     <y>439</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnTextImage</sender>
   <signal>clicked()</signal>
   <receiver>ImageSpecialEffectsQtClass</receiver>
   <slot>openTextImageDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>814</x>
     <y>319</y>
    </hint>
    <hint type="destinationlabel">
     <x>764</x>
     <y>496</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_6</sender>
   <signal>clicked()</signal>
   <receiver>ImageSpecialEffectsQtClass</receiver>
   <slot>doEdgeDetection()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>814</x>
     <y>377</y>
    </hint>
    <hint type="destinationlabel">
     <x>765</x>
     <y>542</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_7</sender>
   <signal>clicked()</signal>
   <receiver>ImageSpecialEffectsQtClass</receiver>
   <slot>doNoLightness()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>814</x>
     <y>348</y>
    </hint>
    <hint type="destinationlabel">
     <x>795</x>
     <y>515</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_8</sender>
   <signal>clicked()</signal>
   <receiver>ImageSpecialEffectsQtClass</receiver>
   <slot>doNoiseReduce()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>814</x>
     <y>406</y>
    </hint>
    <hint type="destinationlabel">
     <x>736</x>
     <y>459</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_9</sender>
   <signal>clicked()</signal>
   <receiver>ImageSpecialEffectsQtClass</receiver>
   <slot>doEdgeSmoothing()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>814</x>
     <y>435</y>
    </hint>
    <hint type="destinationlabel">
     <x>801</x>
     <y>477</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_10</sender>
   <signal>clicked()</signal>
   <receiver>ImageSpecialEffectsQtClass</receiver>
   <slot>doBinarize()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>814</x>
     <y>232</y>
    </hint>
    <hint type="destinationlabel">
     <x>776</x>
     <y>202</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_11</sender>
   <signal>clicked()</signal>
   <receiver>ImageSpecialEffectsQtClass</receiver>
   <slot>doSalience()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>814</x>
     <y>464</y>
    </hint>
    <hint type="destinationlabel">
     <x>748</x>
     <y>527</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_12</sender>
   <signal>clicked()</signal>
   <receiver>ImageSpecialEffectsQtClass</receiver>
   <slot>doLab()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>771</x>
     <y>488</y>
    </hint>
    <hint type="destinationlabel">
     <x>727</x>
     <y>402</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_13</sender>
   <signal>clicked()</signal>
   <receiver>ImageSpecialEffectsQtClass</receiver>
   <slot>undo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>802</x>
     <y>157</y>
    </hint>
    <hint type="destinationlabel">
     <x>796</x>
     <y>172</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>openFile()</slot>
  <slot>saveFile()</slot>
  <slot>doGray()</slot>
  <slot>updateSize(int)</slot>
  <slot>undoZoom()</slot>
  <slot>openBalanceDialog()</slot>
  <slot>openTextImageDialog()</slot>
  <slot>doEdgeDetection()</slot>
  <slot>doNoLightness()</slot>
  <slot>doNoiseReduce()</slot>
  <slot>doEdgeSmoothing()</slot>
  <slot>doBinarize()</slot>
  <slot>doSalience()</slot>
  <slot>doLab()</slot>
  <slot>undo()</slot>
 </slots>
</ui>
